libpod/runtime_pod_infra_linux.go

```diff
@@ -145,6 +146,34 @@ func (r *Runtime) makeInfraContainer(ctx context.Context, p *Pod, imgName, rawIm
 		if len(p.config.InfraContainer.ExitCommand) > 0 {
 			options = append(options, WithExitCommand(p.config.InfraContainer.ExitCommand))
 		}
+		if p.config.Pid.NSMode != "private" && p.config.Pid.NSMode != "" {
+			path := p.config.Pid.Value
+			ns := spec.LinuxNamespace{}
+			pidVal := p.config.Pid.NSMode
+			if pidVal != "" {
+				if pidVal == "container" {
```

Comment:
This validation should be done in the option (`WithPid()` so we only have to do it once and we can reject invalid configurations at creation time, not runtime.

Reply:
Okay @mheon , to streamline this and preserve the originally passed values I might make another InfraContainerConfig value that is an actual `specs.LinuxNameSpace{}`. This will allow me to be able to print the proper mode in podInspect but also not have to parse anything again in `runtime_pod_infra_linux.go`